### ISSUE TITLE
Web Frameworks Team

### ISSUE THREAD

**Issue by wesleytodd** [nodejs/admin#459]
We would like to kick off a new team. The scope of the team would be as follows:

> This WG would be focused around how the core http(1/2/3) and other related apis fit the needs of the web frameworks (express, fastify, restify, hapi, koa etc). It would also give the maintainers a forum to discuss and collaborate on issues which span the projects.

The first things we will need are a team and a repo.  The working name is "Web Server Frameworks", so it would make sense to me to start with `web-server-frameworks` as the repo name (assuming it is not a problem to change the name if we land on something more fitting).

So far the initial list of members are:

@wesleytodd
@dougwilson
@jasnell 
@retrohacker
@mcollina
@Ethan-Arrowood
@ghermeto
@hueniverse
@JozefFlakus
@delvedor
@sheplu
@ayenisholah
@niftylettuce
@fed135 
@donutespresso
@AlexZeitler
@arei
@gireeshpunathil
@RamirezAlex
@drawm
@MarcinHoppe
@Alyssa-Cooper
@antsmartian
@kamilmysliwiec
@ronag
@LoicPoullain 
@mhdawson
@bajtos
@raymondfeng
@blakeembrey

I only listed those who explicitly said they want to be involved in https://github.com/nodejs/TSC/issues/795 or at the Collab summit.

I am sure there are others, so if you see this and are interested please comment and I will add you to the list above, but here are some others who might also want to be:

@linusu
@jonathanong
@dead-horse
@hekike
@hacksparrow

**Comment by hueniverse** [comment]
I am certainly interested in learning more about such effort.

**Comment by JozefFlakus** [comment]
Interesting idea, I'm curious about the topics that will be covered there. As a framework author you can add me as well.

**Comment by wesleytodd** [comment]
For those who follow here, just to be clear, if you express interest I will add you to the list of members above.  If you do not want this, please be clear about your intent.  @JozefFlakus and @hueniverse are both added to the members list!

**Comment by delvedor** [comment]
I'm very interested as well, great initiative! :)

**Comment by sheplu** [comment]
Not a web framework dev but starting to contribute. But using them on serverless most of the time.
So if I can help I will be very interested !

**Comment by MylesBorins** [comment]
I think this is an awesome idea and am happy to help in anyway y'all need to be successful.

I do want to suggest that this effort be framed as a team to start. To become a working group involves creating a charter and being given clear authority over specific work areas of the Node.js project. While I see this effort as massively improving our ecosystem I do not yet see clarity on what exactly the group would like authority over, and as such a team makes more sense while that spins up.

Some historical bits to keep in mind. The `modules team`, is still a team after 2+ years of active work on ESM and is only now [starting to discuss chartering](https://github.com/nodejs/modules/issues/412). The `release working group` started as the `lts team` and [only chartered in 2017](https://github.com/nodejs/Release/pull/223) 2+ years after it's formation.

I mention these two instances not to create any sort of barrier, especially time wise, but more to point out that both of these initiatives were extremely successful in making large / impactful change in Node.js, and it was only at the point that there was no questions regarding ownership (release) or that something significant had shipped (modules) that chartering was actively discussed.

**Comment by Xstoudi** [comment]
@RomainLanz and @thetutlage should be listed in here IMO

**Comment by wesleytodd** [comment]
@MylesBorins I agree and have updated the request.  I am not sure we would ever want specific authority over work in Node.js and I agree that chartering would serve no purpose for us now.  One thing to point out, while this process has been done a few times there are no docs, only docs for starting a WG (see https://github.com/nodejs/TSC/issues/795).

**Comment by ayenisholah** [comment]
I am interested. I love Node.js

**Comment by MylesBorins** [comment]
@wesleytodd I think I found [the documentation](https://github.com/nodejs/admin/blob/master/GITHUB_ORG_MANGEMENT_POLICY.md#teams)

TL;DR

* Make issue to make a new repo (this one)
* Request team to own new repo
  * if no team exists request to make a new one
* In 72 hours if there are no objections team + repo are made.

Seems like you've done all the heavy lifting already! I can help with making the repo + team in 72 hours if there are no objections

**Comment by niftylettuce** [comment]
I'd be interested, actively building [Lad](https://lad.js.org)

**Comment by wesleytodd** [comment]
Also, how do people feel about changing the name to `Server Frameworks Team`?  It was brought to my attention that `Web Frameworks` strongly implies the front-end ones (react, vue, etc), which is not my intended focus.

**Comment by retrohacker** [comment]
A quick sampling of some frameworks home pages:

Express calls itself a web framework
Fastify calls itself a web framework
Restify calls itself a web service framework
Hapi calls itself a framework
Koa calls itself a web framework

I have a preference to keep "web framework" in the name since it seems to be rather standard across the ecosystem. But ultimately don't care too much about what color we paint this shed :smile:

**Comment by wesleytodd** [comment]
Web Server Framework?

**Comment by fed135** [comment]
This is very relevant to my interests!

**Comment by niftylettuce** [comment]
HTTP Framework is pretty simple/focused, and doesn't limit us to just Web since a lot of these also serve as API frameworks.  But then do people consider API part of Web?  What about using one of these frameworks as API for a React Native iOS/Android app?

**Comment by wesleytodd** [comment]
I really dont want to limit to http.  I think protocol doesn't really matter as long as the domain is the server framework.  And for sure api servers fit within this fully.  It looks like atleast a few gave "Web Server Framework Team" the 👍, but does anyone have other ideas (or object to that name)?

**Comment by JozefFlakus** [comment]
I would say that, as far it is visible under `nodejs` organization it carries the domain implicit. If it will stay as `Web Frameworks Team` everyone will recognize the domain well, and it will be easier to remember and operate.

**Comment by DonutEspresso** [comment]
I am interested too.

**Comment by AlexZeitler** [comment]
I would be interested, mostly from a Hypermedia perspective.

**Comment by arei** [comment]
Interested.

**Comment by gireeshpunathil** [comment]
request to add me to the team. my motivations:
 - to help the middleware abstractions of these frameworks are well aligned / stacked up with node core functions
 - consider these frameworks as the prominent node.js workloads, gather usage insights and feed it back to the core

**Comment by RamirezAlex** [comment]
I would love to help :)

**Comment by drawm** [comment]
Count me in. I would love to get involved!

**Comment by MarcinHoppe** [comment]
I will definitely be following this effort and will try to contribute if anything security related pops up.

**Comment by Alyssa-Cooper** [comment]
This is a great initiative. I'd love to contribute!

**Comment by antsmartian** [comment]
This is a great initiative. I'm someone who involve myself with load of web development in my day to day life, would love to contribute in this idea. Count me in.

**Comment by kamilmysliwiec** [comment]
Great idea, I'm happy to help as well ([NestJS](https://github.com/nestjs))

**Comment by mcollina** [comment]
I’m in!

I would also invite @ronag, who has done extensive work on http in core.

**Comment by ronag** [comment]
I'm interested as well.

**Comment by LoicPoullain** [comment]
I'm interested as well! ([FoalTS](https://github.com/FoalTS/foal))

**Comment by mhdawson** [comment]
I'm in as well.

**Comment by bajtos** [comment]
I'd like to join too (https://loopback.io)

**Comment by raymondfeng** [comment]
I'd like to join too - [LoopBack](https://loopback.io).

**Comment by blakeembrey** [comment]
Thanks @wesleytodd, I'd love to be involved.

**Comment by wesleytodd** [comment]
Ok, I have added everyone who has posted.  According to the docs it needs to wait 72 hours without objection, and I posted 3 days ago, so I guess we are good to go!  

Thanks everyone for the interest and once we get the team and repo setup I will post a doodle so we can pick a time for our first meeting.

**Comment by MylesBorins** [comment]
thanks @wesleytodd for pinging the thread. I'll go ahead and spin stuff up.

**Comment by MylesBorins** [comment]
Team created, Names in the list in original post have been added to that team. I've also made the repo and kicked it off with basic a basic README, code of conduct, license, and starter governance (including process for adding new members).

Gonna close this issue so y'all can take convo to the new repo

https://github.com/nodejs/web-server-frameworks
